source/Xunit.Gherkin.Quick/CoreModel/ScenarioOutlineExecutor.cs

```diff
@@ -35,8 +35,9 @@ public ScenarioOutlineExecutor(IFeatureFileRepository featureFileRepository)
                 throw new InvalidOperationException($"Cannot find scenario outline `{scenarioOutlineName}`.");
 
             var gherkinScenario = gherkinScenarioOutline.ApplyExampleRow(exampleName, exampleRowIndex);
+			var gherkinBackground = featureFile.GetBackground();
 
-            var scenario = featureClass.ExtractScenario(gherkinScenario);
+			var scenario = featureClass.ExtractScenario(gherkinScenario, gherkinBackground);			
```

Comment:
I think this calls for a refactoring long run.
If you see line 37, I transform scenario outline into a scenario by calling a (extension) method on the outline (`ApplyExampleRow`). That makes the intent clear (scenario=scenarioOutline+appliedSpecificExampleRowToIt).
Similarly, we could also call after having the background and before this line, something like this:
`gherkinScenario = gherkinScenario.ApplyBackground(gherkinBackground);`
And here we can pass just a single scenario. That same new extension method can be reused in the ScenarioExecutor as well, so also less code duplication.
That should keep the intent clear and reusable, since a scenario with background is just a scenario with more steps prepended to it.
If you have time to go that route, that would be awesome. If not, I will do that when I find time.
Let me know what you do for this comment.
I may be asking for too much for a single PR, but the outcome is definitely worth it.